exercises/react/.meta/plugins.py

```diff
@@ -0,0 +1,20 @@
+def add_callbacks(case):
+    observers = []
+    callbacks = []
+    operations = list(filter(lambda op: op["type"] == "add_callback", case["input"]["operations"]))
+
+    if len(operations) == 0:
+        return '', {}
+
+    sufix = ''
+    if len(operations) > 1:
+        sufix = 'cb{i}_'
+    for i, op in enumerate(operations):
+        callbacks.append(op["name"])
+        observers.append(f'{sufix.format(i=i + 1)}observer')
+
+    assignment_block = f'\n        {", ".join(observers)} = {", ".join(len(callbacks) * ["[]"])}\n'
+    for callback, observer in zip(callbacks, observers):
+        assignment_block += f'        {callback} = self.callback_factory({observer})\n'
```

Comment:
This seems brittle to me because it delegates the newlines and indent to the plugin, but the indentation level particularly is really a concern of the template. Maybe it would be better to return a list of observers and a list of callbacks and have the template handle whitespace?

Reply:
yep, I delegated the indentation to template and I refactored the code.